File: dotnet/src/Api/UseCases/CreateRecipe/Request.cs
using MediatR;

namespace CleanSlice.Api.UseCases.CreateRecipe
{
    public record CreateRecipeRequest : IRequest<CreateRecipeResponse>
    {
        public string Url { get; init; } = null!;

        public string Title { get; init; } = null!;

        public string Name { get; init; } = null!;

        public string Author { get; init; } = null!;
    }
}

File: java/src/main/java/demo/cleanslice/usecases/create_recipe/Request.java
package demo.cleanslice.usecases.create_recipe;

import lombok.Getter;

public class Request {

    @Getter
    private String url;

    @Getter
    private String title;

    @Getter
    private String name;

    @Getter
    private String author;

    public Request(String url, String title, String name, String author) {
        this.url = url;
        this.title = title;
        this.name = name;
        this.author = author;
    }
}
